diff --git a/package.json b/package.json
index 4c4ca3606..b129afadd 100644
--- a/package.json
+++ b/package.json
@@ -80,17 +80,11 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.13.9",
-    "@comagic/core": "^1.17.0",
     "@comagic/crud": "2.60.5",
     "@comagic/flow": "0.6.4",
     "@comagic/reports": "8.6.1",
-    "@comagic/softphone-core": "0.0.0-0.0.44-int0",
-    "@comagic/softphone-widget": "0.0.135-int0",
-    "@comagic/ui": "3.26.1",
-    "@comagic/workplace-utils": "0.0.25-int0",
     "@sentry/react": "^7.73.0",
     "async-mutex": "^0.3.2",
-    "axios": "0.27.2",
     "classnames": "^2.2.6",
     "dayjs": "1.11.11",
     "dompurify": "^2.3.10",
@@ -108,15 +102,10 @@
     "linkifyjs": "4.3.2",
     "lodash.clonedeep": "^4.5.0",
     "madge": "^5.0.1",
-    "magic-ui": "0.0.0-int1",
-    "mobx": "6.6.1",
-    "mobx-react-lite": "3.4.0",
-    "mobx-state-tree": "5.1.8",
     "moment": "2.29.4",
     "nanoid": "^3.2.0",
     "opus-recorder": "^8.0.5",
     "qs": "6.13.0",
-    "react": "18.3.1",
     "react-animate-height": "^2.0.23",
     "react-beautiful-dnd": "^13.1.1",
     "react-color": "2.19.3",
@@ -124,11 +113,7 @@
     "react-csv": "^2.0.3",
     "react-dnd": "^16.0.1",
     "react-dnd-html5-backend": "^16.0.1",
-    "react-dom": "18.3.1",
     "react-hook-form": "7.46.1",
-    "react-intl": "^6.5.5",
-    "react-router": "6.9.0",
-    "react-router-dom": "6.9.0",
     "redux": "^4.0.0",
     "redux-thunk": "^2.3.0",
     "sanitize-html": "2.14.0",
diff --git a/src/modules/chats-distribution/models/scenario/ScenarioStore.ts b/src/modules/chats-distribution/models/scenario/ScenarioStore.ts
index 096114cf9..d13036f26 100644
--- a/src/modules/chats-distribution/models/scenario/ScenarioStore.ts
+++ b/src/modules/chats-distribution/models/scenario/ScenarioStore.ts
@@ -343,7 +343,7 @@ const ScenarioStore = LoaderModel.named('ScenarioStore')
         scenarioListSearch: types.optional(types.string, ''),
         channels: types.optional(types.array(ChannelModel), []),
         scenario: types.optional(ScenarioModel, {}),
-        scenarioHistory: types.array(types.frozen()),
+        scenarioHistory: types.optional(types.array(types.frozen()), []),
         scenarioHistoryUndoIdx: -1,
         lastActiveNodeId: types.frozen<string>('start'),
         returnCode: types.frozen(
diff --git a/src/modules/chats/components/chats/chats-list-panel/components/chat-list/index.tsx b/src/modules/chats/components/chats/chats-list-panel/components/chat-list/index.tsx
index 1250f7d32..1c49226b3 100644
--- a/src/modules/chats/components/chats/chats-list-panel/components/chat-list/index.tsx
+++ b/src/modules/chats/components/chats/chats-list-panel/components/chat-list/index.tsx
@@ -17,24 +17,26 @@ const ChatList: React.FC<ChatListProps> = observer(({ chatStateType, children })
     const { formatMessage: f } = useIntl();
 
     const {
-        chatListStore: { fetchChatListItems, getChatsCountByType, getLoadingByStatus, countersAreLoading },
+        chatListStore,
+        chatListStore: { fetchChatListItems, getChatsCountByType },
     } = useMst();
 
-    const isLoading = useMst().chatListStore.isLoading || countersAreLoading || getLoadingByStatus(chatStateType);
-    const chatsCountByType = getChatsCountByType(chatStateType);
+    const loadingByStatus = () => chatListStore.getLoadingByStatus(chatStateType);
+
+    const isLoading = () => chatListStore.isLoading ||
+        chatListStore.countersAreLoading ||
+        loadingByStatus();
 
     const onScrollToEnd = useCallback(() => {
         void fetchChatListItems(chatStateType);
     }, [chatStateType, fetchChatListItems]);
 
     useEffect(() => {
-        !chatsCountByType && !isLoading && onScrollToEnd();
+        setTimeout(() => !getChatsCountByType(chatStateType) && !isLoading() && onScrollToEnd(), 0);
     }, []);
 
-    const loadingByStatus = getLoadingByStatus(chatStateType);
-
     if (Children.count(children) === 0) {
-        if (loadingByStatus) {
+        if (loadingByStatus()) {
             return undefined;
         }
         return (
@@ -48,7 +50,7 @@ const ChatList: React.FC<ChatListProps> = observer(({ chatStateType, children })
 
     return (
         <div className={classNames(styles.root)}>
-            <InfiniteScroll isLoading={isLoading} onScrollToEnd={onScrollToEnd}>
+            <InfiniteScroll isLoading={isLoading()} onScrollToEnd={onScrollToEnd}>
                 {children}
             </InfiniteScroll>
         </div>
