diff --git a/package.json b/package.json
index 31f2de4..1636533 100644
--- a/package.json
+++ b/package.json
@@ -80,7 +80,6 @@
     "highcharts-react-official": "^3.0.0",
     "history": "^4.10.1",
     "lodash": "^4.17.20",
-    "magic-ui": "git+https://magic-ui:9d_M4ruzxtnTCt__5qs1@gitlab.uis.dev/web/magic_ui.git#v1.14.5",
     "mobx": "^6.3.8",
     "mobx-react-lite": "^3.2.2",
     "mobx-state-tree": "^5.1.0",
diff --git a/src/components/ShadowContent.ts b/src/components/ShadowContent.ts
index dc4e8eb..43d205f 100644
--- a/src/components/ShadowContent.ts
+++ b/src/components/ShadowContent.ts
@@ -14,9 +14,10 @@ export const getShadowRoot = (): HTMLElement => {
 
         parentEl.appendChild(moduleEl);
 
-        moduleEl.attachShadow({ mode: 'open' });
+        //moduleEl.attachShadow({ mode: 'open' });
     }
 
+    return moduleEl;
     return (moduleEl.shadowRoot as unknown) as HTMLElement;
 };
 
diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index 05fcfcd..33fda59 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -93,7 +93,9 @@ export type DependencyInjection = {
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({}, { httpRpc });
+
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}, { httpRpc }));
 
 type RootInstanceSnapshotOut = SnapshotOut<typeof RootStore>;
 
diff --git a/src/models/reports/RootReportStore.ts b/src/models/reports/RootReportStore.ts
index 2a07782..7a34ed9 100644
--- a/src/models/reports/RootReportStore.ts
+++ b/src/models/reports/RootReportStore.ts
@@ -48,7 +48,7 @@ export const RootReportStore = types
         customerHistoryStore: types.optional(CustomerHistoryStore, {}),
     })
     .views(self => {
-        const getReportRouterParams = (pathname = window.location.pathname) => {
+        const getReportRouterParams = (pathname = window.memoryHistory.location.pathname) => {
             const match = matchPath<RouterParams>(pathname, {
                 path: REPORT_PATH,
             });
