diff --git a/package.json b/package.json
index 37fec94..7ce5d77 100644
--- a/package.json
+++ b/package.json
@@ -80,7 +80,6 @@
     "highcharts-react-official": "^3.0.0",
     "history": "^4.10.1",
     "lodash": "^4.17.20",
-    "magic-ui": "git+https://magic-ui:9d_M4ruzxtnTCt__5qs1@gitlab.uis.dev/web/magic_ui.git#v1.15.4",
     "mobx": "^6.3.8",
     "mobx-react-lite": "^3.2.2",
     "mobx-state-tree": "^5.1.0",
diff --git a/src/components/ShadowContent.ts b/src/components/ShadowContent.ts
deleted file mode 100644
index dc4e8eb..0000000
--- a/src/components/ShadowContent.ts
+++ /dev/null
@@ -1,31 +0,0 @@
-import React from 'react';
-import ReactDOM from 'react-dom';
-
-export const getShadowRoot = (): HTMLElement => {
-    const parentEl = document.querySelector('#rootMain');
-
-    if (!parentEl) return;
-
-    let moduleEl = parentEl.querySelector('#analytics');
-
-    if (!moduleEl) {
-        moduleEl = document.createElement('div');
-        moduleEl.setAttribute('id', 'analytics');
-
-        parentEl.appendChild(moduleEl);
-
-        moduleEl.attachShadow({ mode: 'open' });
-    }
-
-    return (moduleEl.shadowRoot as unknown) as HTMLElement;
-};
-
-const ShadowContent: React.FC = ({ children }) => {
-    const shadowRoot = getShadowRoot();
-
-    if (!shadowRoot) return null;
-
-    return ReactDOM.createPortal(children, (shadowRoot as unknown) as Element);
-};
-
-export default ShadowContent;
diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index 05fcfcd..33fda59 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -93,7 +93,9 @@ export type DependencyInjection = {
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({}, { httpRpc });
+
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}, { httpRpc }));
 
 type RootInstanceSnapshotOut = SnapshotOut<typeof RootStore>;
 
diff --git a/src/models/reports/RootReportStore.ts b/src/models/reports/RootReportStore.ts
index 2a07782..7a34ed9 100644
--- a/src/models/reports/RootReportStore.ts
+++ b/src/models/reports/RootReportStore.ts
@@ -48,7 +48,7 @@ export const RootReportStore = types
         customerHistoryStore: types.optional(CustomerHistoryStore, {}),
     })
     .views(self => {
-        const getReportRouterParams = (pathname = window.location.pathname) => {
+        const getReportRouterParams = (pathname = window.memoryHistory.location.pathname) => {
             const match = matchPath<RouterParams>(pathname, {
                 path: REPORT_PATH,
             });
