diff --git a/.env b/.env
index a8ae4714..acdff3a8 100644
--- a/.env
+++ b/.env
@@ -8,6 +8,6 @@ REACT_APP_SITE_SETTINGS_SERVER_URL=
 REACT_APP_ANALYTICS_SERVER_URL=
 REACT_APP_COMAGIC_WEB_URL=https://int0.dev.uis.st
 REACT_APP_PROJECT=
-REACT_APP_MODULE_CHATS=https://chats-int0-chats.uis.st
-REACT_APP_LEADGEN_MODULE=https://leadgen-int0-leadgen-web.uis.st
-REACT_APP_MODULE_SOFTPHONE=https://softphone-int0-softphone.uis.st
+REACT_APP_MODULE_CHATS=https://localhost:8082
+REACT_APP_LEADGEN_MODULE=https://localhost:8084
+REACT_APP_MODULE_SOFTPHONE=https://localhost:8083
diff --git a/config/webpack.config.js b/config/webpack.config.js
index df24e5bb..c85e6844 100644
--- a/config/webpack.config.js
+++ b/config/webpack.config.js
@@ -9,6 +9,7 @@ const MiniCssExtractPlugin = require('mini-css-extract-plugin');
 const CssMinimizerPlugin = require('css-minimizer-webpack-plugin');
 const TerserPlugin = require('terser-webpack-plugin');
 const comagicCoreDeps = require('@comagic/core/package.json');
+const hash = require('string-hash');
 
 const { ModuleFederationPlugin } = webpack.container;
 const deps = require('../package.json').dependencies;
@@ -17,19 +18,23 @@ const NODE_ENV = process.env.NODE_ENV || 'prod';
 const REACT_APP = /^REACT_APP_/i;
 
 const appDirectory = fs.realpathSync(process.cwd());
+const miscDirectory = `${appDirectory}/misc`;
 const dotenv = path.resolve(appDirectory, '.env');
 const themeLessFileName = path.resolve(appDirectory, 'theme', 'theme.less');
 const isProductionMode = NODE_ENV !== 'development';
 const webpackMode = isProductionMode ? 'production' : 'development';
 const localIdentName = isProductionMode ? '[hash:base64]' : '[path][name]__[local]';
 const isBundleAnalyzerPlugin = false;
+const subModules = ['chats', 'employees'];
 
 const lessLoader = {
     loader: 'less-loader',
     options: {
         lessOptions: {
             modifyVars: {
-	        'hack': `true; @import "${themeLessFileName}";`,
+                'hack': `true; @import "${themeLessFileName}"; ${
+                    subModules.map(module => `@import "${miscDirectory}/${module}/theme/theme.less"`).join('; ')
+                }`,
             },
             javascriptEnabled: true,
         },
@@ -184,7 +189,7 @@ const plugins = () => {
             ignoreOrder: true,
         }),
 
-        new ModuleFederationPlugin({
+        /*new ModuleFederationPlugin({
             name: 'shell',
             shared: getSharedDeps(),
             remotes: {
@@ -192,7 +197,11 @@ const plugins = () => {
                 chats: `chats@${process.env.REACT_APP_MODULE_CHATS}/remoteEntry.js`,
                 softphone: `softphone@${process.env.REACT_APP_MODULE_SOFTPHONE}/remoteEntry.js`,
             },
-        }),
+        }),*/
+
+        new webpack.optimize.LimitChunkCountPlugin({
+            maxChunks: 1
+        })
     ];
 
     if (isBundleAnalyzerPlugin) {
@@ -202,6 +211,13 @@ const plugins = () => {
     return pluginsList;
 };
 
+const createAliases = appDirectory => ({
+    '@': path.resolve(appDirectory, './src'),
+    '@models': path.resolve(appDirectory, './src/models'),
+    '@lib': path.resolve(appDirectory, './src/components/lib'),
+    '@hooks': path.resolve(appDirectory, './src/hooks'),
+});
+
 module.exports = {
     entry: './src/index.tsx',
     target: 'web',
@@ -221,6 +237,21 @@ module.exports = {
     module: {
         rules: [
             {
+                test: /\.js$/,
+                exclude: /node_modules(?!\/uis-softphone\/)/,
+                use: [{
+                    loader: 'babel-loader',
+                    options: {
+                        presets: ['@babel/preset-react', '@babel/preset-env'],
+                        comments: false,
+                        plugins: [
+                            '@babel/plugin-proposal-private-methods',
+                            '@babel/plugin-proposal-class-properties',
+                        ]
+                    }
+                }]
+            },
+            {
                 test: /\.(ts|tsx)$/,
                 use: {
                     loader: 'ts-loader',
@@ -271,6 +302,7 @@ module.exports = {
             },
             {
                 test: /\.svg(\?v=\d+\.\d+\.\d+)?$/,
+                exclude: /\/(softphone|uis_webrtc\/src)\/icons\//,
                 use: [
                     {
                         loader: '@svgr/webpack',
@@ -286,17 +318,89 @@ module.exports = {
                     },
                 ],
             },
+            {
+                test: /\/(softphone|uis_webrtc\/src)\/icons\//,
+                use: ({resource}) => {
+                    return [{
+                        loader: 'babel-loader',
+                        options: {
+                            presets: ['@babel/preset-react', '@babel/preset-env'],
+                            comments: false
+                        }
+                    }, {
+                        loader: '@svgr/webpack',
+                        options: {
+                            babel: false,
+                            icon: true,
+                            svgoConfig: {
+                                plugins: [{
+                                    cleanupIDs: {
+                                        prefix: `svg${hash(path.relative(__dirname, resource))}`,
+                                        force: true
+                                    }
+                                }]
+                            }
+                        }
+                    }];
+                }
+            },
         ],
     },
     resolve: {
         extensions: ['.tsx', '.ts', '.js'],
         alias: {
-            '@': path.resolve(appDirectory, './src'),
-            '@models': path.resolve(appDirectory, './src/models'),
-            '@lib': path.resolve(appDirectory, './src/components/lib'),
-            '@hooks': path.resolve(appDirectory, './src/hooks'),
+            ...createAliases(appDirectory),
             '@ant-design/icons/lib/dist$': path.resolve(appDirectory, './src/icons.ts'),
+            'leadgen/Root': `${miscDirectory}/Stub.js`,
+            'employees/Root': 'employees/src/Root.tsx',
+            'softphone/Account': 'softphone/src/Account.tsx',
+            'softphone/Settings': 'softphone/src/Settings.tsx',
+            'softphone/Root': 'softphone/src/Root.tsx',
+            'chats/Root': 'chats/src/Root.tsx',
+            'chats/Account': 'chats/src/components/account/index.tsx',
+            'chats/BasicAccount': 'chats/src/components/account/BasicAccount.tsx',
+            'chats/ReportChatHistoryDrawer': 'chats/src/components/chats/chat-panel/ReportChatHistoryDrawer.tsx',
+            'chats/Menu': 'chats/src/components/main/menu/index.tsx',
         },
+        plugins: [{
+            apply: resolver => resolver.getHook('resolve').tapAsync('RelativeAliasResolvingPlugin', (
+                request,
+                resolveContext,
+                callback
+            ) =>  {
+                const target = resolver.ensureHook('resolve'),
+                    splitted = request.request.split('/');
+
+                const {
+                    directory,
+                    aliases
+                } = subModules.concat(['softphone']).map(module => {
+                    const directory = `${miscDirectory}/${module}`;
+
+                    return {
+                        directory,
+                        aliases: createAliases(directory)
+                    };
+                }).find(({
+                    directory,
+                    aliases
+                }) => request.path.indexOf(directory) === 0 && Object.keys(aliases).some(
+                    alias => splitted[0] == alias
+                )) || {};
+
+                if (!directory) {
+                    callback();
+                    return;
+                }
+
+                splitted[0] = aliases[splitted[0]]
+
+                return resolver.doResolve(target, {
+                    ...request,
+                    request: splitted.join('/') 
+                }, null, resolveContext, callback);
+            })
+        }]
     },
     output: {
         filename: isProductionMode ? 'static/js/[name].[contenthash:8].js' : 'static/js/bundle.js',
diff --git a/package.json b/package.json
index 7b6e2ea3..e4bcadac 100644
--- a/package.json
+++ b/package.json
@@ -50,6 +50,9 @@
     "css-loader": "^5.1.1",
     "css-minimizer-webpack-plugin": "^1.3.0",
     "cypress": "^4.9.0",
+    "dmg-license": "^1.0.9",
+    "electron": "^10.1.5",
+    "electron-builder": "22.10.5",
     "eslint": "^7.28.0",
     "eslint-config-prettier": "^8.3.0",
     "eslint-plugin-cypress": "^2.11.1",
@@ -77,18 +80,22 @@
     "webpack-bundle-analyzer": "^4.4.0",
     "webpack-cli": "^4.5.0",
     "webpack-dev-server": "^3.11.2",
-    "webpack-merge": "^5.7.3",
-    "electron-builder": "22.10.5",
-    "electron": "^10.1.5",
-    "dmg-license": "^1.0.9"
+    "webpack-merge": "^5.7.3"
   },
   "dependencies": {
     "@comagic/core": "git+https://comagic-core:bCb2p2VJ6HxAMxries77@gitlab.uis.dev/web/comagic_app_modules/core.git#0.0.8",
-    "antd": "^3.26.7",
+    "antd": "3.26.7",
+    "auto-launch": "^5.0.5",
     "axios": "^0.19.2",
+    "babel-loader": "^8.0.4",
+    "chats": "./misc/chats",
     "classnames": "^2.2.6",
     "dotenv": "^8.2.0",
     "dotenv-expand": "^5.1.0",
+    "electron-log": "^4.3.0",
+    "electron-store": "^6.0.1",
+    "employees": "./misc/employees",
+    "file-saver": "^2.0.5",
     "formik": "^2.2.9",
     "google-libphonenumber": "^3.2.19",
     "hash-sum": "^2.0.0",
@@ -113,10 +120,8 @@
     "react-resize-detector": "^6.0.0",
     "react-router-dom": "^5.1.2",
     "react-transition-group": "^4.4.2",
+    "softphone": "./misc/softphone",
     "wavesurfer.js": "^4.0.1",
-    "yup": "^0.28.1",
-    "electron-log": "^4.3.0",
-    "electron-store": "^6.0.1",
-    "auto-launch": "^5.0.5"
+    "yup": "^0.28.1"
   }
 }
diff --git a/public/index.html b/public/index.html
index 6143f5e1..1ad2c181 100644
--- a/public/index.html
+++ b/public/index.html
@@ -22,7 +22,24 @@
     })(window, document, 'script', 'dataLayer', 'GTM-TJXG5CR');</script>
     <!-- End Google Tag Manager -->
     <% } %>
-    <script>window.rootConfig = <%= rootConfig %>;</script>
+    <script src="/tests/utils/jasmine/lib/jasmine-3.4.0/jasmine.js"></script>
+    <script src="/tests/utils/jasmine/lib/jasmine-3.4.0/jasmine-html.js"></script>
+    <script>
+        var windowLoadHandler = window.onload;
+    </script>
+    <script src="/tests/utils/jasmine/lib/jasmine-3.4.0/boot.js"></script>
+    <script>
+        window.onload = windowLoadHandler;
+    </script>
+    <script src="/tests/utils/jasmine/lib/jasmine-ajax/lib/mock-ajax.js"></script>
+    <script src="/tests/utils/jasmine/console-reporter.js"></script>
+    <script src="/tests/utils/js-tester.js"></script>
+    <script type="text/javascript">
+        var tests = new JsTester_Tests(new JsTester_Factory());
+        tests.exposeDebugUtils('jsTestDebug');
+    </script>
+    <script src="/tests/utils/require.js"></script>
+    <script src="/tests/tests.js"></script>
 </head>
 <body>
     <% if (htmlWebpackPlugin.options.env.production) { %>
@@ -34,5 +51,6 @@
     <% } %>
     <noscript>You need to enable JavaScript to run this app.</noscript>
     <div id="root"></div>
+    <script src="/tests/utils/tests.js"></script>
 </body>
 </html>
\ No newline at end of file
diff --git a/src/bootstrap.tsx b/src/bootstrap.tsx
index aaf65408..41b85ee5 100644
--- a/src/bootstrap.tsx
+++ b/src/bootstrap.tsx
@@ -8,7 +8,9 @@ import { observer } from 'mobx-react-lite';
 import { CoreProvider } from '@comagic/core';
 import moment from 'moment';
 
-import { Provider, rootStore } from '@models/RootStore';
+import { Provider, rootStore, createRootStore } from '@models/RootStore';
+import { createRootStore as createSoftphoneRootStore } from 'softphone/src/models/RootStore';
+import { createRootStore as createChatsRootStore } from 'chats/src/models/RootStore';
 
 import history from './history';
 import { getAntdLocale, getIntlLocale, getIntlMessages } from './i18n';
@@ -67,4 +69,33 @@ const Root: React.FC = observer(props => {
     );
 });
 
-ReactDOM.render(<Root {...(window.rootConfig || {})}/>, document.getElementById('root'));
+window.application = {
+    run({
+        setHistory = () => null,
+        appName = ''
+    }) {
+        createRootStore();
+        createChatsRootStore();
+        createSoftphoneRootStore();
+        setHistory(history);
+
+        this.exit();
+
+        const container = document.createElement('div');
+        container.id = 'root';
+        document.body.appendChild(container);
+
+        ReactDOM.render(<Root appName={appName} />, container);
+    },
+
+    exit() {
+        const container = document.getElementById('root');
+
+        if (!container) {
+            return;
+        }
+
+        container.firstChild && ReactDOM.unmountComponentAtNode(container);
+        container.remove();
+    }
+};
diff --git a/src/history.ts b/src/history.ts
index 33bf6a02..fe524735 100644
--- a/src/history.ts
+++ b/src/history.ts
@@ -1,6 +1,6 @@
-import { createBrowserHistory } from 'history';
+import { createMemoryHistory } from 'history';
 
-const history = createBrowserHistory();
+const history = window.memoryHistory = createMemoryHistory();
 
 //TODO: может заменить на location.reload()?
 export const reloadPage = (): void => {
diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index 7a7b6f71..0b4bf619 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -1,9 +1,11 @@
 import React from 'react';
 import { applySnapshot, getSnapshot, Instance, SnapshotOut, types } from 'mobx-state-tree';
 import axios, { AxiosRequestConfig } from 'axios';
+import moment from 'moment';
 import Highcharts from 'highcharts';
 import patternFill from 'highcharts/modules/pattern-fill';
 import HighchartsCustomEvents from 'highcharts-custom-events';
+import * as fileSaver from 'file-saver';
 
 import { GOD_MODE } from '@/config';
 import { httpRpc, httpRpcAuth } from '@/rpc';
@@ -79,16 +81,16 @@ const projectsData: Record<string, ProjectData> = {
 
 type RequestToRefresh = (nextResolve: boolean) => void;
 
+const getIpcRender = () => (
+    window.rootConfig?.appName == 'softphone' &&
+    ((window.require || (() => null))('electron')?.ipcRenderer)
+) || {
+    on: () => null,
+    send: () => null
+};
+
 const createSoftphoneStoreCreatingHandler = rootStore => {
     let handleStoreCreating = store => {
-        const ipcRenderer = (
-            window.rootConfig?.appName == 'softphone' &&
-            ((window.require || (() => null))('electron')?.ipcRenderer)
-        ) || {
-            on: () => null,
-            send: () => null
-        };
-
         [{
             channelName: 'checkautolauncher',
             paramName: 'isStartApp',
@@ -98,11 +100,9 @@ const createSoftphoneStoreCreatingHandler = rootStore => {
             paramName: 'isStartMinimize',
             setParamValue: 'setStartMinimize'
         }].forEach(({channelName, paramName, setParamValue}) =>
-            ipcRenderer.on(channelName, (event, data) => store[setParamValue]((data || {})[paramName])));
-
-        ipcRenderer.on('logfilepath', (event, data) => rootStore.setLogFilePath(data.logFilePath));
-        ipcRenderer.send('app-ready');
+            getIpcRender().on(channelName, (event, data) => store[setParamValue]((data || {})[paramName])));
 
+        getIpcRender().send('app-ready');
         handleStoreCreating = () => null;
     };
 
@@ -112,6 +112,7 @@ const createSoftphoneStoreCreatingHandler = rootStore => {
 export const RootStore = types
     .model('RootStore', {
         isGodMode: types.optional(types.boolean, isGodMode()),
+        isLogDownloading: types.optional(types.boolean, false),
         modalResponseErrorStore: types.optional(ModalResponseErrorStore, {}),
         authStore: types.optional(AuthStore, {}),
         accountStore: types.optional(AccountStore, {}),
@@ -125,10 +126,7 @@ export const RootStore = types
         numberStore: types.optional(NumberStore, {}),
         numberPoolStore: types.optional(NumberPool, {}),
         numberPoolNumbersStore: types.optional(NumberPoolNumbersStore, {}),
-        marketplaceStore: types.optional(MarketplaceStore, {}),
-        logFilePath: types.maybe(types.string),
-        message: types.maybe(types.string),
-        title: types.maybe(types.string)
+        marketplaceStore: types.optional(MarketplaceStore, {})
     })
     .views(self => ({
         get projectData() {
@@ -194,7 +192,7 @@ export const RootStore = types
 
             setAuthHeaders();
 
-            axios.interceptors.response.use(
+            window.axiosInterceptors = axios.interceptors.response.use(
                 response => {
                     if (response.data.error) {
                         const {
@@ -272,11 +270,27 @@ export const RootStore = types
 
         let setSoftphoneStore = () => null;
 
+        const downloadLog = () => {
+            self.isLogDownloading = true;
+            getIpcRender().send('collect_logs');
+        };
+
+        const handleLogsCollected = data => {
+            self.isLogDownloading = false;
+
+            fileSaver.saveAs(new Blob([data], {
+                type: 'application/zip'
+            }), `Softphone.${moment().format('YYYYMMDD.HHmmss.SSS')}.log.zip`)
+        };
+
         const afterCreate = () => {
+            window.axiosInterceptors !== undefined && axios.interceptors.response.eject(window.axiosInterceptors);
             initialState = getSnapshot(self);
 
             setHighchartsOptions();
             setHttpRequestsDefault();
+
+            getIpcRender().on('logs_collected', (event, data) => self.handleLogsCollected(data));
             setSoftphoneStore = createSoftphoneStoreCreatingHandler(self);
         };
 
@@ -284,23 +298,8 @@ export const RootStore = types
             applySnapshot(self, initialState);
         };
 
-        const setLogFilePath = value => (self.logFilePath = value);
-
-        const closeMessage = () => {
-            self.title = '';
-            self.message = '';
-        };
-
-        const setMessage = (title, message) => {
-            self.title = title;
-            self.message = message;
-        };
-
         self.fileLogger = {
-            downloadLog: () => self.setMessage(
-                'Log path',
-                self.logFilePath
-            ),
+            downloadLog: () => self.downloadLog(),
 
             log: (...messages) => window.require('electron').remote.require('electron-log').info(
                 messages.map(
@@ -312,9 +311,8 @@ export const RootStore = types
         return {
             afterCreate,
             resetStore,
-            setMessage,
-            closeMessage,
-            setLogFilePath,
+            downloadLog,
+            handleLogsCollected,
             setSoftphoneStore: store => setSoftphoneStore(store) 
         };
     });
@@ -325,7 +323,8 @@ export type DependencyInjection = {
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({}, { httpRpc, httpRpcAuth });
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}, { httpRpc, httpRpcAuth }));
 
 type RootInstanceSnapshotOut = SnapshotOut<typeof RootStore>;
 
diff --git a/src/models/auth/AuthStore.ts b/src/models/auth/AuthStore.ts
index ca1e21ed..70b68e72 100644
--- a/src/models/auth/AuthStore.ts
+++ b/src/models/auth/AuthStore.ts
@@ -105,7 +105,7 @@ const removeToken = (skipLocalStorage = false) => {
 
 export const AuthStore = Loader.named('AuthStore')
     .props({
-        isAuthorized: types.optional(types.boolean, Boolean(getToken())),
+        isAuthorized: types.optional(types.boolean, false),
         isTokenRefreshing: types.optional(types.boolean, false),
         windowId: types.optional(types.string, generateTraceId('local')),
     })
@@ -204,6 +204,8 @@ export const AuthStore = Loader.named('AuthStore')
             isTokenRefreshing = false;
             refreshingCall = null;
 
+            setAuthorized(Boolean(getToken()));
+
             window.addEventListener('doLogoutUser', (event: CustomEvent) => {
                 const eventWindowId = event.detail;
                 if (eventWindowId !== authStore.windowId) {
diff --git a/src/rpc/httpRpc.ts b/src/rpc/httpRpc.ts
index 742f44e3..8a47a0d6 100644
--- a/src/rpc/httpRpc.ts
+++ b/src/rpc/httpRpc.ts
@@ -27,10 +27,11 @@ const buildMode: string = process.env.REACT_APP_BUILD_MODE;
 const baseUrl: string =
     buildMode === 'e2e_test' ? process.env.REACT_APP_BASE_URL_FOR_E2E : process.env.REACT_APP_BASE_URL;
 
-let cancelTokenSource: CancelTokenSource = axios.CancelToken.source();
+let cancelTokenSource: CancelTokenSource = null;
+const getCancelTokenSource = () => cancelTokenSource || (cancelTokenSource = axios.CancelToken.source());
 
 export const cancelPendingRequests = (message?: string): void => {
-    cancelTokenSource.cancel(message);
+    getCancelTokenSource().cancel(message);
     cancelTokenSource = axios.CancelToken.source();
 };
 
@@ -47,7 +48,7 @@ const httpRpc: HttpRpc = async <T>(method: string, params = {}, headers = {}, ur
         data: rpcParams,
         url: `${url}?method=${method}`,
         headers,
-        cancelToken: cancelTokenSource.token,
+        cancelToken: getCancelTokenSource().token,
     };
 
     try {
