diff --git a/package.json b/package.json
index 403b917..2262e40 100644
--- a/package.json
+++ b/package.json
@@ -21,7 +21,6 @@
     "@types/node": "^14.14.37",
     "@types/react": "^17.0.2",
     "@types/react-dom": "^17.0.1",
-    "@types/react-router-dom": "^5.1.6",
     "@types/uuid": "^8.3.1",
     "@types/yup": "^0.29.11",
     "@typescript-eslint/eslint-plugin": "^4.16.1",
@@ -54,7 +53,7 @@
   "dependencies": {
     "@babel/runtime": "^7.13.9",
     "@comagic/core": "^0.6.4",
-    "antd": "^4.13.0",
+    "antd": "3.26.7",
     "async-mutex": "^0.3.2",
     "axios": "^0.21.1",
     "classnames": "^2.2.6",
@@ -64,7 +63,6 @@
     "formik": "^2.2.9",
     "history": "^4.10.1",
     "madge": "^5.0.1",
-    "magic-ui": "^1.19.15",
     "mobx": "^6.3.8",
     "mobx-devtools-mst": "^0.9.30",
     "mobx-react-lite": "^3.2.2",
@@ -73,10 +71,8 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "react-intl": "^5.22.0",
-    "react-router-dom": "^5.2.0",
     "short-uuid": "^4.2.0",
     "uuid": "^8.3.2",
-    "yup": "^0.32.9",
-    "@uiscom/uis_webrtc": "git+https://uis_webrtc:glpat-ydssS-Qi8NFPN7dmjkcp@gitlab.uis.dev/web/uis_webrtc.git#local-multitab"
+    "yup": "^0.32.9"
   }
 }
diff --git a/src/App.tsx b/src/App.tsx
index 01b5463..7585277 100644
--- a/src/App.tsx
+++ b/src/App.tsx
@@ -45,11 +45,11 @@ const App: React.FC = observer(() => {
         };
     });
 
-    return (
+    return [false, true].some(value => value === isAuthorized) ? (
         <div className="cm-app-module">
             <AppRoute />
         </div>
-    );
+    ) : null;
 });
 
 export default App;
diff --git a/src/history.ts b/src/history.ts
index 564e8ec..e156b1f 100644
--- a/src/history.ts
+++ b/src/history.ts
@@ -1,5 +1,2 @@
-import { createBrowserHistory } from 'history';
-
-const history = createBrowserHistory();
-
+const history = window.memoryHistory;
 export default history;
diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index d71456e..00dd413 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -67,7 +67,8 @@ export const RootStore = types
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({});
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}));
 
 if (process.env.NODE_ENV === 'development') {
     window.chatsRootStore = rootStore;
diff --git a/src/models/auth/AuthStore.ts b/src/models/auth/AuthStore.ts
index 933fdb3..8e6f31d 100644
--- a/src/models/auth/AuthStore.ts
+++ b/src/models/auth/AuthStore.ts
@@ -26,7 +26,7 @@ const AUTH_KEY: string = process.env.REACT_APP_AUTH_COOKIE || '$REACT_APP_AUTH_C
 
 export const AuthStore = types
     .model('AuthStore', {
-        isAuthorized: types.optional(types.boolean, Boolean(false)),
+        isAuthorized: types.maybe(types.boolean),
         windowId: types.optional(types.string, generateTraceId('local')),
     })
     .views(() => ({
