diff --git a/package.json b/package.json
index 94b71d5..19e2d1d 100644
--- a/package.json
+++ b/package.json
@@ -55,7 +55,7 @@
   "dependencies": {
     "@babel/runtime": "^7.13.9",
     "@comagic/core": "git+https://comagic-core:bCb2p2VJ6HxAMxries77@gitlab.uis.dev/web/comagic_app_modules/core.git#0.5.2",
-    "antd": "^4.13.0",
+    "antd": "3.26.7",
     "async-mutex": "^0.3.2",
     "axios": "^0.21.1",
     "classnames": "^2.2.6",
@@ -65,7 +65,6 @@
     "formik": "^2.2.9",
     "history": "^4.10.1",
     "madge": "^5.0.1",
-    "magic-ui": "git+https://magic-ui:9d_M4ruzxtnTCt__5qs1@gitlab.uis.dev/web/magic_ui.git#1.1.55",
     "mobx": "^6.3.8",
     "mobx-devtools-mst": "^0.9.30",
     "mobx-react-lite": "^3.2.2",
diff --git a/src/App.tsx b/src/App.tsx
index 5f054bd..1386b7e 100644
--- a/src/App.tsx
+++ b/src/App.tsx
@@ -43,7 +43,7 @@ const App: React.FC = observer(() => {
         };
     });
 
-    return (
+    return [false, true].some(value => value === isAuthorized) ? (
         <div className="cm-app-module">
             <Switch>
                 <Route path="/login" render={props => <LoginRoute {...props} isAuthorized={isAuthorized} />} />
@@ -52,7 +52,7 @@ const App: React.FC = observer(() => {
                 </AuthProtectedRoute>
             </Switch>
         </div>
-    );
+    ) : null;
 });
 
 export default App;
diff --git a/src/history.ts b/src/history.ts
index 564e8ec..e156b1f 100644
--- a/src/history.ts
+++ b/src/history.ts
@@ -1,5 +1,2 @@
-import { createBrowserHistory } from 'history';
-
-const history = createBrowserHistory();
-
+const history = window.memoryHistory;
 export default history;
diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index 772a2d0..b94e54f 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -87,7 +87,8 @@ export const RootStore = types
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({});
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}));
 
 if (process.env.NODE_ENV === 'development') {
     window.chatsRootStore = rootStore;
diff --git a/src/models/auth/AuthStore.ts b/src/models/auth/AuthStore.ts
index f470280..29d9152 100644
--- a/src/models/auth/AuthStore.ts
+++ b/src/models/auth/AuthStore.ts
@@ -24,7 +24,7 @@ type Auth = {
 
 export const AuthStore = types
     .model('AuthStore', {
-        isAuthorized: types.optional(types.boolean, Boolean(false)),
+        isAuthorized: types.maybe(types.boolean),
         windowId: types.optional(types.string, generateTraceId('local')),
     })
     .views(self => ({
