diff --git a/package.json b/package.json
index 7ff3ecb..68c3d36 100644
--- a/package.json
+++ b/package.json
@@ -54,7 +54,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.13.9",
-    "antd": "^4.13.0",
+    "antd": "^3.26.7",
     "async-mutex": "^0.3.2",
     "axios": "^0.21.1",
     "classnames": "^2.2.6",
diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index 772a2d0..b94e54f 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -87,7 +87,8 @@ export const RootStore = types
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({});
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}));
 
 if (process.env.NODE_ENV === 'development') {
     window.chatsRootStore = rootStore;
