diff --git a/package.json b/package.json
index a0bd309..2c4d755 100644
--- a/package.json
+++ b/package.json
@@ -22,8 +22,6 @@
     "@types/dompurify": "^2.3.3",
     "@types/htmlparser2": "^3.10.3",
     "@types/node": "^14.14.37",
-    "@types/react": "^18.0.21",
-    "@types/react-dom": "^18.0.6",
     "@types/uuid": "^8.3.1",
     "@types/yup": "^0.29.11",
     "@typescript-eslint/eslint-plugin": "^4.16.1",
@@ -55,8 +53,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.13.9",
-    "@comagic/core": "0.0.0-int0",
-    "antd": "^3.26.7",
+    "antd": "3.26.7",
     "async-mutex": "^0.3.2",
     "axios": "^0.21.1",
     "classnames": "^2.2.6",
@@ -67,16 +64,11 @@
     "formik": "^2.2.9",
     "html-react-parser": "^3.0.1",
     "madge": "^5.0.1",
-    "magic-ui": "^2.3.0",
     "mobx": "^6.6.1",
     "mobx-devtools-mst": "^0.9.30",
     "mobx-react-lite": "^3.4.0",
     "mobx-state-tree": "^5.1.6",
     "moment": "^2.29.1",
-    "react": "^18.2.0",
-    "react-dom": "^18.2.0",
-    "react-intl": "^6.1.1",
-    "react-router-dom": "^6.4.0",
     "short-uuid": "^4.2.0",
     "uuid": "^8.3.2",
     "yup": "^0.32.9"
diff --git a/src/App.tsx b/src/App.tsx
index 43db4d8..6a8f91e 100644
--- a/src/App.tsx
+++ b/src/App.tsx
@@ -9,7 +9,7 @@ import './App.less';
 
 const App: React.FC = observer(() => {
     const {
-        authStore: { updateAuthState },
+        authStore: { updateAuthState, isAuthorized },
         websocketStore,
         chatListStore,
         chatChannelsStore,
@@ -42,11 +42,11 @@ const App: React.FC = observer(() => {
         };
     });
 
-    return (
+    return [false, true].some(value => value === isAuthorized) ? (
         <div className="cm-app-module">
             <AppRoute />
         </div>
-    );
+    ) : null;
 });
 
 export default App;
diff --git a/src/components/chats/chat-panel/styles.less b/src/components/chats/chat-panel/styles.less
index b1e30f9..70bc523 100644
--- a/src/components/chats/chat-panel/styles.less
+++ b/src/components/chats/chat-panel/styles.less
@@ -1,3 +1,5 @@
+@import '~antd/dist/antd.css';
+
 .@{p}chat-panel, .@{p}chat-like-panel {
   .@{p}chat-panel-quote {
     display: grid;
diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index f0d453d..d6ce0a6 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -144,7 +144,8 @@ export const RootStore = types
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({});
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}));
 
 if (process.env.NODE_ENV === 'development') {
     window.chatsRootStore = rootStore;
diff --git a/src/models/account/AccountStore.ts b/src/models/account/AccountStore.ts
index 57ad272..166fd36 100644
--- a/src/models/account/AccountStore.ts
+++ b/src/models/account/AccountStore.ts
@@ -1,4 +1,4 @@
-import { flow, Instance, types, cast } from 'mobx-state-tree';
+import { flow, Instance, types, cast, getSnapshot } from 'mobx-state-tree';
 import { httpRpc, DataApiResponseResult } from '@/httpRpc';
 import { intl } from '@/i18n';
 
@@ -55,12 +55,18 @@ export const AccountStore = types
     })
     .views(self => ({
         getOtherEmployeeChatsAccess() {
-            const permission = [...self.account.permissions].find(item => item[0] === 'other_employee_chats_access');
-            return permission && permission[1] ? permission[1].is_select : false;
+            const permission = getSnapshot(self.account.permissions).other_employee_chats_access;
+            return permission?.is_select || false;
         },
         getTagsPermissions() {
-            const tagsPermissions = [...self.account.permissions].find(item => item[0] === 'tag_management')[1];
-            return tagsPermissions;
+            return (
+                getSnapshot(self.account.permissions).tag_management || {
+                    is_delete: false,
+                    is_insert: false,
+                    is_select: false,
+                    is_update: false,
+                }
+            );
         },
     }))
     .actions(self => {
diff --git a/src/models/auth/AuthStore.ts b/src/models/auth/AuthStore.ts
index 17e5a68..ea4743d 100644
--- a/src/models/auth/AuthStore.ts
+++ b/src/models/auth/AuthStore.ts
@@ -27,7 +27,7 @@ const AUTH_KEY: string = process.env.REACT_APP_AUTH_COOKIE || '$REACT_APP_AUTH_C
 
 export const AuthStore = types
     .model('AuthStore', {
-        isAuthorized: types.optional(types.boolean, Boolean(false)),
+        isAuthorized: types.maybe(types.boolean),
         windowId: types.optional(types.string, generateTraceId('local')),
     })
     .views(() => ({
