diff --git a/package.json b/package.json
index f40d08d..34dc9ab 100644
--- a/package.json
+++ b/package.json
@@ -22,9 +22,6 @@
     "@types/dompurify": "^2.3.3",
     "@types/htmlparser2": "^3.10.3",
     "@types/node": "^14.14.37",
-    "@types/react": "^17.0.2",
-    "@types/react-dom": "^17.0.1",
-    "@types/react-router-dom": "^5.1.6",
     "@types/uuid": "^8.3.1",
     "@types/yup": "^0.29.11",
     "@typescript-eslint/eslint-plugin": "^4.16.1",
@@ -56,8 +53,7 @@
   },
   "dependencies": {
     "@babel/runtime": "^7.13.9",
-    "@comagic/core": "^0.6.4",
-    "antd": "^4.13.0",
+    "antd": "3.26.7",
     "async-mutex": "^0.3.2",
     "axios": "^0.21.1",
     "classnames": "^2.2.6",
@@ -69,19 +65,13 @@
     "history": "^4.10.1",
     "html-react-parser": "^3.0.1",
     "madge": "^5.0.1",
-    "magic-ui": "./magic_ui",
     "mobx": "^6.3.8",
     "mobx-devtools-mst": "^0.9.30",
     "mobx-react-lite": "^3.2.2",
     "mobx-state-tree": "^5.0.5",
     "moment": "^2.29.1",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2",
-    "react-intl": "^5.22.0",
-    "react-router-dom": "^5.2.0",
     "short-uuid": "^4.2.0",
     "uuid": "^8.3.2",
-    "yup": "^0.32.9",
-    "@uiscom/uis_webrtc": "git+https://uis_webrtc:glpat-ydssS-Qi8NFPN7dmjkcp@gitlab.uis.dev/web/uis_webrtc.git#local-multitab"
+    "yup": "^0.32.9"
   }
 }
diff --git a/src/App.tsx b/src/App.tsx
index 43db4d8..6a8f91e 100644
--- a/src/App.tsx
+++ b/src/App.tsx
@@ -9,7 +9,7 @@ import './App.less';
 
 const App: React.FC = observer(() => {
     const {
-        authStore: { updateAuthState },
+        authStore: { updateAuthState, isAuthorized },
         websocketStore,
         chatListStore,
         chatChannelsStore,
@@ -42,11 +42,11 @@ const App: React.FC = observer(() => {
         };
     });
 
-    return (
+    return [false, true].some(value => value === isAuthorized) ? (
         <div className="cm-app-module">
             <AppRoute />
         </div>
-    );
+    ) : null;
 });
 
 export default App;
diff --git a/src/history.ts b/src/history.ts
index 564e8ec..e156b1f 100644
--- a/src/history.ts
+++ b/src/history.ts
@@ -1,5 +1,2 @@
-import { createBrowserHistory } from 'history';
-
-const history = createBrowserHistory();
-
+const history = window.memoryHistory;
 export default history;
diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index 4457fa5..05b069d 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -120,7 +120,8 @@ export const RootStore = types
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({});
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}));
 
 if (process.env.NODE_ENV === 'development') {
     window.chatsRootStore = rootStore;
diff --git a/src/models/auth/AuthStore.ts b/src/models/auth/AuthStore.ts
index 933fdb3..8e6f31d 100644
--- a/src/models/auth/AuthStore.ts
+++ b/src/models/auth/AuthStore.ts
@@ -26,7 +26,7 @@ const AUTH_KEY: string = process.env.REACT_APP_AUTH_COOKIE || '$REACT_APP_AUTH_C
 
 export const AuthStore = types
     .model('AuthStore', {
-        isAuthorized: types.optional(types.boolean, Boolean(false)),
+        isAuthorized: types.maybe(types.boolean),
         windowId: types.optional(types.string, generateTraceId('local')),
     })
     .views(() => ({
diff --git a/src/utils/index.tsx b/src/utils/index.tsx
index af48314..fc10827 100644
--- a/src/utils/index.tsx
+++ b/src/utils/index.tsx
@@ -1,8 +1,8 @@
 import React from 'react';
 import { renderToStaticMarkup } from 'react-dom/server';
 import cn from 'classnames';
-import parse, { attributesToProps, domToReact, DOMNode } from 'html-react-parser';
-import DOMPurify from 'dompurify';
+//import parse, { attributesToProps, domToReact, DOMNode } from 'html-react-parser';
+//import DOMPurify from 'dompurify';
 import moment from 'moment';
 import short from 'short-uuid';
 import { intl } from '@/i18n';
@@ -113,8 +113,15 @@ type Options = {
     'max-age'?: number;
 };
 
+const isSoftphone = () => window.rootConfig?.appName == 'softphone',
+    prepareName = name => encodeURIComponent(name.replace(/([.$?*|{}()[\]\\/+^])/g, '\\$1'));
+
 export function getCookie(name: string): string {
-    const matches = new RegExp('(?:^|; )' + name.replace(/([.$?*|{}()[\]\\/+^])/g, '\\$1') + '=([^;]*)').exec(
+    if (isSoftphone()) {
+        return window.getElectronCookiesManager().getItem(name);
+    }
+
+    const matches = new RegExp('(?:^|; )' + prepareName(name) + '=([^;]*)').exec(
         document.cookie
     );
 
@@ -122,6 +129,11 @@ export function getCookie(name: string): string {
 }
 
 export function setCookie(name: string, value: string, options: Options = {}): void {
+    if (isSoftphone()) {
+        window.getElectronCookiesManager().setItem(name, value);
+        return;
+    }
+
     options = {
         path: '/',
         ...options,
@@ -131,7 +143,7 @@ export function setCookie(name: string, value: string, options: Options = {}): v
         options.expires = options.expires.toUTCString();
     }
 
-    let updatedCookie = encodeURIComponent(name) + '=' + encodeURIComponent(value);
+    let updatedCookie = prepareName(name) + '=' + encodeURIComponent(value);
 
     let optionKey: keyof Options;
     for (optionKey in options) {
@@ -139,7 +151,7 @@ export function setCookie(name: string, value: string, options: Options = {}): v
 
         const optionValue = options[optionKey];
         if (optionValue !== true) {
-            updatedCookie += '=' + (optionValue as string);
+            updatedCookie += '=' + optionValue;
         }
     }
 
@@ -147,6 +159,11 @@ export function setCookie(name: string, value: string, options: Options = {}): v
 }
 
 export function deleteCookie(name: string, options: Options = {}): void {
+    if (isSoftphone()) {
+        window.getElectronCookiesManager().removeItem(name);
+        return;
+    }
+
     setCookie(name, '', {
         'max-age': -1,
         ...options,
@@ -190,6 +207,7 @@ function generateUrl(url: string): string {
 }
 
 export function parseLinks(text: string): React.ReactNode[] {
+    return text;
     text = `${text || ''}`
         .split(' ')
         .map(string => (isUrl(string) ? generateUrl(string) : string))
