diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index 73408a7..dced2c4 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -713,4 +713,5 @@ export const RootStore = types
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({});
\ No newline at end of file
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}));
\ No newline at end of file
