diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index b6ca744..66e84a3 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -800,4 +800,5 @@ export type RootInstance = Instance<typeof RootStore> & {
 
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({});
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}));
