diff --git a/package.json b/package.json
index 5374789..83687da 100644
--- a/package.json
+++ b/package.json
@@ -53,7 +53,7 @@
   "dependencies": {
     "@babel/runtime": "^7.13.9",
     "@comagic/core": "0.0.0-int0",
-    "@uiscom/uis_webrtc": "0.0.0-int0",
+    "@uiscom/uis_webrtc": "0.0.0-int2",
     "async-mutex": "^0.3.2",
     "axios": "^0.21.1",
     "classnames": "^2.2.6",
diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index aa311e2..8da548b 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -220,7 +220,7 @@ export const RootStore = types
                     };
 
                     const isSoftphoneAvailable = () =>
-                        state.accessGranted && state.mounted && state.token;
+                        !!(state.accessGranted && state.mounted && state.token);
 
                     const authorize = () =>
                         isSoftphoneAvailable() &&
@@ -413,7 +413,7 @@ export const RootStore = types
                 });
 
                 call.name == 'toggleCallInfoCollapsedness' && self.setNotLarge();
-                call.name == 'setSettings' && broadcast('settings_fetched', ...call.args);
+                call.name == 'onSettingsProcessed' && broadcast('settings_fetched', call.args[0]);
 
                 if (call.name == 'handleEvent') {
                     const data = (data => {
@@ -717,4 +717,5 @@ export const RootStore = types
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({});
\ No newline at end of file
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}));
\ No newline at end of file
