diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index 91a06ca..bd6ad87 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -15,6 +15,7 @@ import {
     openUrl,
     openContact,
     isAbleToOpenContact,
+    isElectron,
 } from '@/utils';
 
 import {
@@ -25,13 +26,7 @@ import {
 } from '@comagic/softphone-widget';
 
 import { getResponse } from '@comagic/softphone-core/src/utils';
-
-import {
-    eventBus,
-    isElectron,
-    ModuleAvailability,
-    handleModuleAvailabilityGetterFetched
-} from '@comagic/core';
+import { eventBus } from '@comagic/core';
 
 import InfopinFeedbackStore from '@models/feedback/FeedbackWebStore';
 import FeedbackWebStore from '@models/feedback/FeedbackWebStore';
@@ -81,7 +76,6 @@ export const RootStore = types
         promperStore: types.optional(PromperStore, {}),
         callHistoryStore: types.optional(CallHistoryStore, {}),
         callStatsStore: types.optional(CallStatsStore, {}),
-        moduleAvailability: types.optional(ModuleAvailability, {}),
     })
     .views(self => ({
         i18n: key => intlMessages(self.locale)[key],
@@ -322,11 +316,6 @@ export const RootStore = types
                     const accessGranted = getSoftphoneAvailability(accountStore),
                         account = accountStore.toJSON()?.account;
 
-                    handleModuleAvailabilityGetterFetched({
-                        moduleAvailability: self.moduleAvailability,
-                        accountStore,
-                    })('chats', import('chats/isWorkplaceAllowed'));
-
                     accessGranted && accountFetchedHandlers.forEach(handler => handler(accountStore));
                     updateState({ accessGranted });
 
@@ -674,4 +663,5 @@ export const RootStore = types
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({});
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}));
