diff --git a/package.json b/package.json
index b987f3e..c047c3d 100644
--- a/package.json
+++ b/package.json
@@ -55,7 +55,7 @@
   "dependencies": {
     "nanoid": "2.1.11",
     "@babel/runtime": "^7.13.9",
-    "antd": "^4.13.0",
+    "antd": "3.26.7",
     "async-mutex": "^0.3.2",
     "axios": "^0.21.1",
     "classnames": "^2.2.6",
@@ -79,6 +79,6 @@
     "short-uuid": "^4.2.0",
     "uuid": "^8.3.2",
     "yup": "^0.32.9",
-    "uis-softphone": "0.0.157"
+    "uis-softphone": "./misc/sip_lib"
   }
 }
diff --git a/src/models/RootStore.ts b/src/models/RootStore.ts
index 1ab983e..2f66b1c 100644
--- a/src/models/RootStore.ts
+++ b/src/models/RootStore.ts
@@ -92,7 +92,7 @@ export const RootStore = types
                         case STATUSES_MANAGER_CREATED:
                             action.statusesManager.on(
                                 'employeeChanged',
-                                ({status_id}) => self.employeesStore.setEmployee({status_id})
+                                ({status_id}) => status_id !== undefined && self.employeesStore.setEmployee({status_id})
                             );
  
                             break;
@@ -250,4 +250,5 @@ export const RootStore = types
 export type RootInstance = Instance<typeof RootStore>;
 export const RootStoreContext = React.createContext<null | RootInstance>(null);
 export const Provider = RootStoreContext.Provider;
-export const rootStore = RootStore.create({});
+export let rootStore;
+export const createRootStore = () => (rootStore = RootStore.create({}));
