diff --git a/config/session.js b/config/session.js
index 090da36..3acf7a8 100644
--- a/config/session.js
+++ b/config/session.js
@@ -25,12 +25,12 @@ module.exports.session = {
 //      database: 'electron_release_server'
 //  }
 
-//  // uncomment if you use sails-pg-session
-//  postgresql: {
-//      adapter: 'sails-pg-session',
-//      host: 'localhost',
-//      user: 'electron_release_server_user',
-//      password: 'MySecurePassword',
-//      database: 'electron_release_server'
-//  }
+  // uncomment if you use sails-pg-session
+  postgresql: {
+      adapter: 'sails-pg-session',
+      host: 'localhost',
+      user: 'electron_release_server_user',
+      password: 'cZSNa6Qc0zdqtljZZ08bMZNJrrTK0ory0De8qlENuqvD31XVGtXIeGadPqmLgHj',
+      database: 'electron_release_server'
+  }
 };
diff --git a/package.json b/package.json
index d544de1..76e1c4f 100644
--- a/package.json
+++ b/package.json
@@ -4,10 +4,11 @@
   "version": "1.5.2",
   "description": "A version server for hosting and serving the your electron desktop app releases.",
   "dependencies": {
-    "async": "^2.5.0",
+    "async": "^2.6.3",
     "bluebird": "^3.4.6",
     "bower": "^1.8.0",
     "compare-versions": "^3.1.0",
+    "sass": "^1.33.0",
     "db-migrate": "^0.11.6",
     "db-migrate-pg": "^1.0.0",
     "express-useragent": "^1.0.4",
@@ -24,7 +25,7 @@
     "grunt-contrib-uglify-es": "^3.3.0",
     "grunt-contrib-watch": "~1.0.0",
     "grunt-sails-linker": "~1.0.4",
-    "grunt-sass": "^2.0.0",
+    "grunt-sass": "^3.1.0",
     "grunt-sync": "~0.6.2",
     "grunt-wiredep": "^3.0.1",
     "include-all": "~2.0.0",
diff --git a/tasks/config/sass.js b/tasks/config/sass.js
index dc3cc87..3e1f5a4 100644
--- a/tasks/config/sass.js
+++ b/tasks/config/sass.js
@@ -3,6 +3,7 @@ module.exports = function(grunt) {
   grunt.config.set('sass', {
     dev: {
       options: {
+          implementation: require('sass')
         // style: 'compressed',
       },
       files: [{
